tests/test_zero_data_parallel/test_shard_model_v2.py

```diff
@@ -31,30 +31,52 @@ def run_fwd_bwd(model, data, label, criterion, enable_autocast=False):
         loss.backward()
 
 
+# with no criterion
+def run_fwd_bwd_no_criterion(model, data, label, enable_autocast=False):
+    model.train()
+    with torch.cuda.amp.autocast(enabled=enable_autocast):
+        loss = model(data, label)
+    if isinstance(model, ShardedModelV2):
+        model.backward(loss)
+    else:
+        loss.backward()
+
+
 def run_dist(rank, world_size, port):
     colossalai.launch(config=CONFIG, rank=rank, world_size=world_size, host='localhost', port=port, backend='nccl')
-    test_models = ['repeated_computed_layers', 'resnet18']
+
+    test_models = ['bert']
+    # repeated_computed_layers resnet18
+    shard_strategy = TensorShardStrategy()
     for model_name in test_models:
         get_components_func = non_distributed_component_funcs.get_callable(model_name)
-        shard_strategy = TensorShardStrategy()
         model, train_dataloader, test_dataloader, optimizer, criterion = get_components_func()
-        model = model().half().cuda()
+        model = model(checkpoint=True).half().cuda()
         zero_model = ShardedModelV2(copy.deepcopy(model), shard_strategy)
         if dist.get_world_size() > 1:
             model = DDP(model)
 
         for i, (data, label) in enumerate(train_dataloader):
             if i > 2:
                 break
-            data, label = data.half().cuda(), label.cuda()
-            run_fwd_bwd(model, data, label, criterion, False)
-            run_fwd_bwd(zero_model, data, label, criterion, False)
+
+            if criterion is None:
+                data, label = data.cuda(), label.cuda()
+                run_fwd_bwd_no_criterion(model, data, label, False)
+                run_fwd_bwd_no_criterion(zero_model, data, label, False)
+            else:
+                # FIXME() data can be interger!
+                data, label = data.half().cuda(), label.cuda()
```

Comment:
@ver217  you should check the type of data before convert it to half()...

Reply:
I will do this in next MR.